src/api/controllers/security.ts

```diff
@@ -113,15 +121,91 @@ export const login = (req: express.Request, res: express.Response) => {
             status: 'failed',
             message: 'Invalid credential. Please try again.',
           });
+          res.end();
         }
-        res.end();
       })
       .catch((e) => {
-        logger.error('got exception while login', e);
+        logger.error(e, 'got exception while login');
         res.status(500).json({
           status: 'failed',
           message: 'Internal error',
         });
+        res.end();
+      });
+  } catch (err) {
+    res.status(500).json({
+      status: 'error',
+      message: 'Internal Server Error',
+    });
+    res.end();
+  }
+};
+
+export const serverLogin = (req: express.Request, res: express.Response) => {
+  try {
+    if (!IsSecurityEnabled()) {
+      res
+        .status(200)
+        .json({
+          status: 'succeed',
+          message: 'security disabled',
+        })
+        .end();
+      return;
+    }
```

Comment:
I would issue a warning in the logs somewhere at startup if no security would be enabled.

Reply:
Fair point, will update.